extensions/cornerstone/src/panels/PanelSegmentation.tsx

```diff
@@ -145,17 +168,41 @@ export default function PanelSegmentation({ children }: withAppTypes) {
     };
   });
 
+  // Update the map of last selected segmentation IDs for the active viewport id.
+  const activeSegmentationInfo = segmentationsWithRepresentations.find(
+    info => info.representation?.active
+  );
+
+  if (activeSegmentationInfo) {
+    const typeToSegmentationIdMap =
+      selectedSegmentationIdByViewportAndType.get(activeViewportId) ?? new Map<string, string>();
+
+    typeToSegmentationIdMap.set(
+      activeSegmentationInfo.representation.type,
+      activeSegmentationInfo.segmentation.segmentationId
+    );
+
+    selectedSegmentationIdByViewportAndType.set(activeViewportId, typeToSegmentationIdMap);
+  }
+
   // Common props for SegmentationTable
   const tableProps = {
     disabled,
     data: segmentationsWithRepresentations,
     mode: segmentationTableMode,
-    title: 'Segmentations',
+    title: `${segmentationRepresentationType ? `${segmentationRepresentationType} ` : ''}Segmentations`,
     exportOptions,
     disableEditing,
     onSegmentationAdd,
     showAddSegment,
     renderInactiveSegmentations: handlers.getRenderInactiveSegmentations(),
+    segmentationRepresentationType,
+    selectedSegmentationIdForType:
+      segmentationRepresentationType !== undefined
+        ? selectedSegmentationIdByViewportAndType
+            .get(activeViewportId)
+            ?.get(segmentationRepresentationType)
+        : activeSegmentationInfo?.segmentation.segmentationId,
```

Comment:
why do we need selectedSegmentationIdForType, shouldn't it always show the acgtive segmentation for the viewport? This is really tricky, whenever the seg is shown in the multi tab means we should actually change the active segmentation as well when we switch tab (that is another reason why i hate the multi tab design)